pkg/controller/populators/populator-base.go

```diff
@@ -225,17 +226,14 @@ type updatePVCAnnotationsFunc func(pvc, pvcPrime *corev1.PersistentVolumeClaim)
 
 func (r *ReconcilerBase) updatePVCWithPVCPrimeAnnotations(pvc, pvcPrime *corev1.PersistentVolumeClaim, updateFunc updatePVCAnnotationsFunc) error {
 	pvcCopy := pvc.DeepCopy()
-	if phase, ok := pvcPrime.Annotations[cc.AnnPodPhase]; ok {
-		cc.AddAnnotation(pvcCopy, cc.AnnPodPhase, phase)
-	}
-	if ready, ok := pvcPrime.Annotations[cc.AnnPodReady]; ok {
-		cc.AddAnnotation(pvcCopy, cc.AnnPodReady, ready)
-	}
-	if restarts, ok := pvcPrime.Annotations[cc.AnnPodRestarts]; ok {
-		cc.AddAnnotation(pvcCopy, cc.AnnPodRestarts, restarts)
-	}
-	if preallocation, ok := pvcPrime.Annotations[cc.AnnPreallocationApplied]; ok {
-		cc.AddAnnotation(pvcCopy, cc.AnnPreallocationApplied, preallocation)
+	for k, v := range pvcPrime.GetAnnotations() {
+		if strings.Contains(k, common.CDIAnnKey) &&
+			!strings.Contains(k, cc.AnnImmediateBinding) &&
+			!strings.Contains(k, cc.AnnPopulatorKind) &&
+			!strings.Contains(k, "upload") &&
+			!strings.Contains(k, "import") {
+			cc.AddAnnotation(pvcCopy, k, v)
```

Comment:
Would it be best to simply have a whitelist of allowed annotations?  What's your opinion?

Reply:
it seems there are too much allowed annotations comparing to annotations that shouldn't be added.. and that's how I missed a bunch, I think its easier to know which we don't want to add

Reply:
Too much?  Could probably count all the annotations we care about on two hands?  Maybe three?  They're all defined and unlikely to change.  Would be nice to have one place (or maybe one place for each source) that states "these are the annotations we care about for datavolumes".  I also clearly have a preference for whitelists over blacklists.